File: src/Enterprise/VB.NET/patterns/Microkernel.vb
Imports System
Imports System.Collections.Generic
Friend Module MicrokernelExample
    Friend Function Run() As Boolean
        Dim plugins As New Dictionary(Of String,Func(Of Integer,Integer)) From {{"double",Function(x)x*2},{"square",Function(x)x*x}}
        Return plugins("double")(4)=8 AndAlso plugins("square")(4)=16
    End Function
End Module

File: src/Enterprise/C#/patterns/Microkernel.cs
using System;
using System.Collections.Generic;
namespace Genkidama.PatternExamples;
public static class MicrokernelExample { public static bool Run(){var p=new Dictionary<string,Func<int,int>>{{"double",x=>x*2},{"square",x=>x*x}};return p["double"](4)==8&&p["square"](4)==16;} }
